src/Microsoft.AspNet.Mvc.DataAnnotations/Internal/MvcDataAnnotationsMvcOptionsSetup.cs

```diff
@@ -12,15 +15,26 @@ namespace Microsoft.AspNet.Mvc.DataAnnotations.Internal
     /// </summary>
     public class MvcDataAnnotationsMvcOptionsSetup : ConfigureOptions<MvcOptions>
     {
-        public MvcDataAnnotationsMvcOptionsSetup()
-            : base(ConfigureMvc)
+        public MvcDataAnnotationsMvcOptionsSetup(
+           IServiceProvider serviceProvider)
+            : base(options => ConfigureMvc(options, serviceProvider))
         {
         }
 
-        public static void ConfigureMvc(MvcOptions options)
+        public static void ConfigureMvc(
+            MvcOptions options,
+            IServiceProvider serviceProvider)
         {
+            var dataAnnotationLocalizationOptions =
+                serviceProvider.GetRequiredService<IOptions<MvcDataAnnotationsLocalizationOptions>>();
```

Comment:
This can get called even outside of locaization in which case, service provider will not have `IOptions<MvcDataAnnotationsLocalizationOptions>` and `IStringLocalizerFactory` services. If I add them to constructor, I get `Unable to resolve service errors`.